packages/meditrak-server/src/routes/exportSurveyResponses/exportSurveyResponses.js

```diff
@@ -143,45 +149,54 @@ export async function exportSurveyResponses(req, res) {
-              comparisonType: 'whereBetween',
-              args: [
-                [
-                  new Date(moment(startDate).subtract(1, 'day')),
-                  new Date(moment(endDate).add(1, 'day')),
-                ],
+        const surveyResponseFindConditions = {
+          survey_id: currentSurvey.id,
+        };
+        if (startDate && endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparisonType: 'whereBetween',
+            args: [
+              [
+                new Date(moment(startDate).subtract(1, 'day')),
+                new Date(moment(endDate).add(1, 'day')),
               ],
-            };
-          } else if (startDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '>=',
-              comparisonValue: new Date(startDate),
-            };
-          } else if (endDate) {
-            surveyResponseFindConditions.submission_time = {
-              comparator: '<=',
-              comparisonValue: new Date(endDate),
-            };
-          }
-          const surveyResponses = await models.surveyResponse.find(
-            surveyResponseFindConditions,
-            sortAndLimitSurveyResponses,
-          );
+            ],
+          };
+        } else if (startDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '>=',
+            comparisonValue: new Date(startDate),
+          };
+        } else if (endDate) {
+          surveyResponseFindConditions.submission_time = {
+            comparator: '<=',
+            comparisonValue: new Date(endDate),
+          };
+        }
+        const entityIdsGroup = entities.map(entity => entity.id);
+        surveyResponseFindConditions.entity_id = entityIdsGroup;
 
-          for (
-            let surveyResponseIndex = 0;
-            surveyResponseIndex < surveyResponses.length;
-            surveyResponseIndex++
-          ) {
-            await processSurveyResponse(surveyResponses[surveyResponseIndex], entity);
-          }
+        const surveyResponses = await models.surveyResponse.find(
+          surveyResponseFindConditions,
+          sortAndLimitSurveyResponses,
+        );
+        const nestedEntitiesById = keyBy(entities, 'id');
+        // Fetch all answers of all survey responses for further use in 'processSurveyResponse()'.
+        const surveyResponseIds = surveyResponses.map(response => response.id);
+        const answers = await findAnswersInSurveyResponse(
+          models,
+          surveyResponseIds,
+          {},
+          { columns: [{ [`${TYPES.SURVEY_RESPONSE}.id`]: 'survey_response.id' }], sort: [] },
+        );
+        const nestedAnswers = groupBy(answers, 'survey_response.id');
+        console.log(nestedAnswers);
+        for (const response of surveyResponses) {
+          await processSurveyResponse(
+            response,
+            nestedEntitiesById[response.entity_id],
+            nestedAnswers,
```

Comment:
I assume we can pass the answers for the current survey directly (similarly to how we directly pass the entity one line above). In this case, you'll have to refactor `processSurveyResponse` a bit

```suggestion
            answersByResponseId[response.id],
```

Reply:
Yeah because we use `processSurveyResponse` at another place here, but I agree we can refactor that as well, then it will look better.